### ISSUE TITLE
Custom width on a UINavigationController segue programmatically

### ISSUE THREAD

**Issue by n3d1117** [SwiftKickMobile/SwiftMessages#313]
Hello, awesome library!

I'm trying to use `SwiftMessagesSegue` with a `UITableViewController` inside a `UINavigationController` using `.centered` layout. So far works fine:

```
let nav = UINavigationController(rootViewController: MyTableViewController())
let segue = SwiftMessagesSegue(identifier: nil, source: self, destination: nav)
segue.configure(layout: .centered)
segue.messageView.backgroundHeight = 250
segue.perform()
```

```
class MyTableViewController: UITableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        navigationItem.leftBarButtonItem = UIBarButtonItem(title: "Cancel", style: .plain, target: self, action: #selector(dismissAnimated))

        // other tableview stuff
    }

    @objc fileprivate func dismissAnimated() {
        dismiss(animated: true)
    }

}
```
![ss1](https://user-images.githubusercontent.com/11541888/57647515-6c721880-75c3-11e9-8e81-51106ccc057f.png)

However I noticed that the modal goes full width even in landscape and on iPads, making it look way too stretched. Is there any way I can set a max width?

![ss2](https://user-images.githubusercontent.com/11541888/57647497-65e3a100-75c3-11e9-9e49-7837ff401369.png)
![ss3](https://user-images.githubusercontent.com/11541888/57647498-65e3a100-75c3-11e9-9918-52d337bd7dab.png)

Thanks!

**Comment by wtmoose** [comment]
I'm working on better defaults for iPad width, which should be available within the next week or so.

As a quick fix, you could try adding a max width constraint to `segue.messageView.backgroundView`.

**Comment by n3d1117** [comment]
Hello,
Thanks for the quick reply! I tried the fix you mentioned:

`segue.messageView.backgroundView.widthAnchor.constraint(lessThanOrEqualToConstant: 600).isActive = true`

But it conflicts with other constraints:

![la](https://user-images.githubusercontent.com/11541888/57650970-6f253b80-75cc-11e9-9321-57ac70b0c08f.png)

**Comment by wtmoose** [comment]
Can you try the head of `master` and let me know if it works for you? It has improved default sizing on iPad and fixes the Auto Layout error you had. Here are the updated readme notes on sizing:

>### Sizing
>
>`SwiftMessagesSegue` provides static default view controller sizing based on device.
>However, it is recommended that you specify sizing appropriate for your content using
>one of the following methods.
>  1. Define sufficient width and height constraints in your view controller.
>  1. Set `preferredContentSize` (a.k.a "Use Preferred Explicit Size" in Interface Builder's attribute inspector). Zeros are ignored, e.g. `CGSize(width: 0, height: 350)` only affects the height.
>  1. Add explicit width and/or height constraints to `segue.messageView.backgroundView`.  
>
>Note that `Layout.topMessage` and `Layout.bottomMessage` are always full screen width. For other layouts, the there is >a maximum 500pt width on iPad (regular horizontal size class) at 950 priority, which can be overridden by adding higher-priority constraints.

**Comment by n3d1117** [comment]
Just tested, it's working now @wtmoose! Great addition to the library, I think you can close this. 
Thank you!

**Comment by wtmoose** [comment]
Fixed in 7.0.0.
